python/tvm/contrib/hexagon/build.py

```diff
@@ -378,13 +239,16 @@ def __init__(
         if not rpc_info.get("workspace_base"):
             rpc_info["workspace_base"] = self.ANDROID_HEXAGON_TEST_BASE_DIR
         self._serial_number = serial_number
+        assert self._serial_number != ""
+
         adb_socket = rpc_info["adb_server_socket"] if rpc_info["adb_server_socket"] else "tcp:5037"
         self._adb_device_sub_cmd = ["adb", "-L", adb_socket, "-s", self._serial_number]
         self.forwarded_ports_ = []
         self._hexagon_debug = hexagon_debug
         self._clear_logcat = clear_logcat
         self._sysmon_profile = sysmon_profile
         self._sysmon_process = None
+        rpc_info["device_key"] = HEXAGON_REMOTE_DEVICE_KEY + "." + self._serial_number
```

Comment:
Each test worker is supposed to use a pair of (android_serial_number, RPC session). But after each worker registers the RPS server to the tracker, it might pickup RPC session with an RPC server which is not related to the same android_serial_number. By adding `self._serial_number` to the rpc_key, each session is forced to use the same RPC server